platform/posix/ota_pal/source/ota_pal_posix.c

```diff
@@ -626,23 +648,30 @@ OtaPalStatus_t otaPal_ResetDevice( OtaFileContext_t * const C )
  */
 OtaPalImageState_t otaPal_GetPlatformImageState( OtaFileContext_t * const C )
 {
-    FILE * pPlatformImageState;
+    FILE * pPlatformImageState = NULL;
     OtaImageState_t eSavedAgentState = OtaImageStateUnknown;
     OtaPalImageState_t ePalState = OtaPalImageStateUnknown;
-    char imageStateFile[ OTA_FILE_PATH_LENGTH_MAX ];
+    char imageStateFile[ OTA_FILE_PATH_LENGTH_MAX ] = { 0 };
+    char * pFileName = NULL;
 
-    /* POSIX port using standard library */
-    /* coverity[misra_c_2012_rule_21_6_violation] */
-    int res = snprintf( imageStateFile, OTA_FILE_PATH_LENGTH_MAX, "%s/%s", getenv( "PWD" ), "PlatformImageState.txt" );
+    ( void ) C;
 
-    assert( res >= 0 );
-    ( void ) res; /* Suppress the unused variable warning when assert is off. */
+    /* Get current directory. */
+    pFileName = getcwd( imageStateFile, OTA_FILE_PATH_LENGTH_MAX );
 
-    ( void ) C;
+    if( pFileName == NULL )
+    {
+        LogError( ( "Failed to get current working directory: %s", strerror( errno ) ) );
+    }
+    else
+    {
+        /* Add the filename . */
+        strncat( imageStateFile, OTA_PLATFORM_IMAGE_STATE_FILE, sizeof( OTA_PLATFORM_IMAGE_STATE_FILE ) );
```

Comment:
See my earlier comment – you're not verifying that `imageStateFile` is of sufficient size to hold both the current working directory and `OTA_PLATFORM_IMAGE_STATE_FILE` before creating the fully-qualified pathname here.  See my earlier suggestion as to how this might be done.

Reply:
Thanks, extracted file path generation into `getFilePathFromCWD()` and checked if the buffer is large enough to hold filepath + filename + `\` before performing `strncat()`